### PR TITLE
[FIX] AttributeError

### PR DESCRIPTION
**Pull request by rven**
AttributeError: 'functools.partial' object has no attribute '__name__'

### CODE REVIEW

**Comment by legalsylvain** [comment]
Hi @rven. It seems a duplicate of https://github.com/OCA/server-env/issues/151 Don't you think ? 
If yes, could you close your issue. 
thanks !

**Comment by rven** [comment]
> Hi @rven. It seems a duplicate of #151 Don't you think ? If yes, could you close your issue. thanks !

@legalsylvain This is about the same issue, but in fact of waiting for Odoo to fix this, it can also be fixed in the way I do in this PR!
@simahawk What do you think?
This was working fine now https://github.com/OCA/storage/pull/254
image

**Comment by simahawk** [comment]
I'm working on a fix on #151

File: server_environment/models/server_env_mixin.py [hunk 1]
```diff
@@ -332,8 +332,14 @@ def _server_env_transform_field_to_read_from_env(self, field):
...
         inverse_method_name = "_inverse_server_env_%s" % field.name
-        inverse_method = partialmethod(type(self)._inverse_server_env, field.name)
-        setattr(type(self), inverse_method_name, inverse_method)
+        inverse_method = partial(
+            type(self)._inverse_server_env, self=self, field_name=field.name
+        )
+        setattr(
+            type(self),
+            inverse_method_name,
+            update_wrapper(inverse_method, type(self)._inverse_server_env),
```
Comment (simahawk) [inline server_environment/models/server_env_mixin.py:341]: hmm weird... I've tried the same fix w/ update_wrapper but it didn't work...
Ah but you are not using `partialmethod` anymore.... :thinking:

Comment (simahawk) [reply server_environment/models/server_env_mixin.py:341]: I need to double check.

**Review by simahawk — CHANGES_REQUESTED** [review]
Just keeping on hold

File: server_environment/models/server_env_mixin.py (hunk 1 above)
Comment (simahawk) [reply server_environment/models/server_env_mixin.py:341]: Also, this solution won't set the right name but the name of the wrapped func. I tried overriding `__name__` but that won't work, at least on v14

Comment (rven) [reply server_environment/models/server_env_mixin.py:341]: Works well in v16.0
image

**Comment by simahawk** [comment]
@rven can you pls have a look at https://github.com/OCA/server-env/pull/153?
I don't think is correct to use `partial` instead of `partialmethod` and in any case `update_wrapper` won't propagate the right name.

**Comment by rven** [comment]
> @rven can you pls have a look at #153? I don't think is correct to use `partial` instead of `partialmethod` and in any case `update_wrapper` won't propagate the right name.

It is indeed not correct, because the record(self) is also not provided well!

**Pull request closed without merging**